src/custom/state/enhancedTransactions/reducer.ts

```diff
@@ -43,72 +48,91 @@ export const initialState: EnhancedTransactionState = {}
         }
-        const txs = transactions[chainId] ?? {}
-        txs[hash] = {
-          hash,
-          hashType,
-          addedTime: now(),
-          from,
-          summary,
-
-          // Operations
-          approval,
-          presign,
+      )
+
+      .addCase(clearAllTransactions, (transactions, { payload: { chainId } }) => {
+        if (!transactions[chainId]) return
+        transactions[chainId] = {}
+      })
+
+      .addCase(checkedTransaction, (transactions, { payload: { chainId, hash, blockNumber } }) => {
+        const tx = transactions[chainId]?.[hash]
+        if (!tx) {
+          return
         }
-        transactions[chainId] = txs
-      }
-    )
-
-    .addCase(clearAllTransactions, (transactions, { payload: { chainId } }) => {
-      if (!transactions[chainId]) return
-      transactions[chainId] = {}
-    })
-
-    .addCase(checkedTransaction, (transactions, { payload: { chainId, hash, blockNumber } }) => {
-      const tx = transactions[chainId]?.[hash]
-      if (!tx) {
-        return
-      }
-      if (!tx.lastCheckedBlockNumber) {
-        tx.lastCheckedBlockNumber = blockNumber
-      } else {
-        tx.lastCheckedBlockNumber = Math.max(blockNumber, tx.lastCheckedBlockNumber)
-      }
-    })
-
-    .addCase(finalizeTransaction, (transactions, { payload: { hash, chainId, receipt } }) => {
-      const tx = transactions[chainId]?.[hash]
-      if (!tx) {
-        return
-      }
-      tx.receipt = receipt
-      tx.confirmedTime = now()
-    })
-
-    .addCase(cancelTransaction, (transactions, { payload: { chainId, hash } }) => {
-      if (!transactions[chainId]?.[hash]) {
-        console.error('Attempted to cancel an unknown transaction.')
-        return
-      }
-      const allTxs = transactions[chainId] ?? {}
-      delete allTxs[hash]
-    })
-
-    .addCase(replaceTransaction, (transactions, { payload: { chainId, oldHash, newHash } }) => {
-      if (!transactions[chainId]?.[oldHash]) {
-        console.error('Attempted to replace an unknown transaction.')
-        return
-      }
-      const allTxs = transactions[chainId] ?? {}
-      allTxs[newHash] = { ...allTxs[oldHash], hash: newHash, addedTime: new Date().getTime() }
-      delete allTxs[oldHash]
-    })
+        if (!tx.lastCheckedBlockNumber) {
+          tx.lastCheckedBlockNumber = blockNumber
+        } else {
+          tx.lastCheckedBlockNumber = Math.max(blockNumber, tx.lastCheckedBlockNumber)
+        }
+      })
+
+      .addCase(finalizeTransaction, (transactions, { payload: { hash, chainId, receipt } }) => {
+        const tx = transactions[chainId]?.[hash]
+        if (!tx) {
+          return
+        }
+        tx.receipt = receipt
+        tx.confirmedTime = now()
+      })
+
+      .addCase(cancelTransaction, (transactions, { payload: { chainId, hash } }) => {
+        if (!transactions[chainId]?.[hash]) {
+          console.error('Attempted to cancel an unknown transaction.')
```

Comment:
above you're throwing and here you're logging and not throwing (better)

Reply:
Yep, this is our code (Ramiro's), the throw comes from uniswap code. Anyways, is good point so changing the orginal